atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/LockStore.java

```diff
@@ -0,0 +1,155 @@
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetRequest;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.Value;
+
+/**
+ * LockStore manages {@link LockEntry} objects, specifically for the "Deletion Lock" (to be taken out by backup and
+ * sweep).
+ *
+ * The PERSISTED_LOCKS_TABLE contains exactly one element, either LOCK_OPEN or (lock taken for some REASON), giving
+ * rise to a state machine where we hop between the central "OPEN" state and some "taken because REASON" state:
+ *
+ * "taken for SWEEP (uuid1)" <- - - - -> OPEN < - - - -> "taken for BACKUP (uuid2)"
+ *
+ * acquireLock(REASON) attempts to move the lock state from "OPEN" to "taken because REASON", returning a LockEntry
+ * that holds a unique identifier for that occasion of taking the lock.
+ *
+ * releaseLock(LockEntry) attempts to move the state from "taken because REASON" to "OPEN", but only succeeds if the
+ * LockEntry currently stored matches the one passed in, ensuring that nobody released the lock when we weren't looking.
+ *
+ * Upon creating the PERSISTED_LOCKS_TABLE, we attempt to enter this state machine by populating the table.
+ * If we fail to do this, it's because someone else also created the table, and populated it before we did.
+ * This is actually OK - all we care about is that we're in the state machine _somewhere_.
+ */
+public final class LockStore {
+    private static final Logger log = LoggerFactory.getLogger(LockStore.class);
+
+    private static final String ROW_NAME = "DeletionLock";
+    private final KeyValueService keyValueService;
+
+    public static final LockEntry LOCK_OPEN = ImmutableLockEntry.builder()
+            .rowName(ROW_NAME)
+            .lockId("-1")
+            .reason("Available")
+            .build();
+
+    private LockStore(KeyValueService kvs) {
+        this.keyValueService = kvs;
+    }
+
+    public static LockStore create(KeyValueService kvs) {
+        kvs.createTable(AtlasDbConstants.PERSISTED_LOCKS_TABLE, AtlasDbConstants.GENERIC_TABLE_METADATA);
+        LockStore lockStore = new LockStore(kvs);
+        ensurePersistedLocksTableIsPopulated(kvs, lockStore);
+        return lockStore;
+    }
+
+    private static void ensurePersistedLocksTableIsPopulated(KeyValueService kvs, LockStore lockStore) {
+        if (lockStore.allLockEntries().isEmpty()) {
+            populateTable(kvs);
+        }
+    }
+
+    @VisibleForTesting
+    void populateTable() {
```

Comment:
I dont like this method created being only for testing. We are not testing code that is being run by doing this, we can either make `ensurePersistedLocksTableIsPopulated` call `populateTable()` or make `populateTable(KeyValueService kvs)` visibleForTesting.